models/balancer/ethereum/balancer_v1_ethereum_pools_tokens_weights.sql

```diff
@@ -0,0 +1,101 @@
+{{
+    config(
+        schema = 'balancer_v1_ethereum',
+        alias='pools_tokens_weights',
+        post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "balancer",
+                                    \'["metacrypto", "jacektrocinski"]\') }}'
+    ) 
+}}
+
+WITH events AS (
+    -- Binds
+    SELECT
+        bind.call_block_number AS block_number,
+        tx.index,
+        bind.call_trace_address,
+        bind.contract_address AS pool,
+        bind.token,
+        bind.denorm
+    FROM {{ source('balancer_v1_ethereum', 'BPool_call_bind') }} bind
+    INNER JOIN {{ source('ethereum', 'transactions') }} tx ON tx.hash = bind.call_tx_hash 
+    WHERE bind.call_success = TRUE
```

Comment:
Since `ethereum.transaction` is used here, we need to use a stricter incremental filter for performance as it is a large table.
```suggestion
    WHERE bind.call_success = TRUE
        {% if not is_incremental() %}
        AND bind.call_block_time >= '{{bind_start_date}}'
        AND tx.block_time >= '{{bind_start_date}}'
        {% endif %}
        {% if is_incremental() %}
        AND bind.call_block_time >= date_trunc("day", now() - interval '1 week')
        AND tx.block_time >= date_trunc("day", now() - interval '1 week')
        {% endif %}
```